### ISSUE TITLE
Unsupported class version number [53.0] (maximum 52.0, Java 1.8)

### ISSUE THREAD

**Issue by schildbach** [bcgit/bc-java#512]
When using version 1.61 in an Android project, I get `Unsupported class version number [53.0] (maximum 52.0, Java 1.8)` on the class `module-info.class`. Class version 53.0 is Java9. There is no such requirement documented in the release notes. This is a regression from BC 1.60, no such problems with that.

**Comment by bcgit** [comment]
Try the latest beta: https://www.bouncycastle.org/betas - it's now using the multi-version jar format. In theory whatever you're using will not find the class as it won't know to look there.

**Comment by schildbach** [comment]
Sorry I didn't get back to this. I have the same problem with bcprov-jdk15on version 1.62.

**Comment by peterdettman** [comment]
Are you using Android Gradle Plugin to build? See https://github.com/bcgit/bc-java/issues/519#issuecomment-492660858 for a possible solution if so.

**Comment by schildbach** [comment]
> Are you using Android Gradle Plugin to build? See [#519 (comment)](https://github.com/bcgit/bc-java/issues/519#issuecomment-492660858) for a possible solution if so.

Yes, I'm using version 2.3.3 of that plugin. I can't upgrade because Google broke build reproducability for later versions. Also, later versions require later Gradle versions which are not yet available in Debian and Ubuntu.

**Comment by peterdettman** [comment]
Are you using our jar signature in any way, e.g. the JCE provider? If not, just remove META-INF/version and all the META-INF signature files from the release jar, and AGP should stop complaining.

We're looking at providing separate release jars without the module-info (and without multi-release code for the jars that use it). I assume for build reproducibility you want to begin with an official release jar?

**Comment by schildbach** [comment]
As far as I know, we don't use the JCE mechanism and instead instantiate and use BCs crypto classes explicitly.

What's an "official release jar"? Is the `org.bouncycastle:bcprov-jdk15on` Gradle/Maven dependency in some way not official? It's kind of documented on https://www.bouncycastle.org/latest_releases.html though.

Can I maybe go back to the JDK14 release? Does that include the Java9 class too?

**Comment by peterdettman** [comment]
We wouldn't recommend going back to JDK14, but no it doesn't have the Java9 stuff.

By "official release jar" I meant the jars that we build and sign for a given release (they go up on our website first, and the same jars are then sent to maven).

What I was saying was that you could manually strip out the META-INF/version and the META-INF signature files from the problem jar yourself, which should work fine if you don't need the signature for JCE usage.

On the other hand, you would presumably prefer us to build such a jar ourselves as part of our next release. We are looking at doing that, but it is a busy time!

**Comment by schildbach** [comment]
Oh ok thanks, understood. Yes, for build reproducibility such a modification of your original JAR would need to be built into our build process. Probably doable (build.gradle is essentially Groovy), but hacky. On your end, it would probably only require

```
jar {     
   exclude('META-INF/versions/9/module-info.class')     
}
```

**Comment by schildbach** [comment]
Side note: I tried `org.bouncycastle:bcprov-jdk14:1.61` instead of the `jdk15on` artifact and it seems to work fine, although Dex throws lots of warnings about having used an old compiler. Version 1.62 seems to have been forgotten when publishing to Maven central / JCenter; it can't be resolved.

**Comment by bcgit** [comment]
Try this jar: https://downloads.bouncycastle.org/misc/bcprov-jdk15to18-1.62.jar

If it works for you we'll make it part of future releases. If it does work, would you post a note to dev-crypto as well (bit stretched for time)? It seems the whole versions things is not playing out anything like it was supposed to. Thanks.

**Comment by schildbach** [comment]
@bcgit The jar works well.

**Comment by peterdettman** [comment]
We're hoping for a 1.63 release in the first week of September. It will include a jdk15to18 jar.

**Comment by schildbach** [comment]
@peterdettman The first week slowly approaching its end, is 1.63 still expected soon?

**Comment by peterdettman** [comment]
Yes, it should be a matter of a few days now.

**Comment by bcgit** [comment]
It's now imminent. Just to be sure would you try: https://downloads.bouncycastle.org/misc/bcprov-jdk15to18-163.jar and let me know how you go with it?

**Comment by schildbach** [comment]
I did some smoke test with that JAR in my app and tested Bitcoin signing and scrypt. Seems to work fine!

**Comment by peterdettman** [comment]
Fixed in 1.63.

**Comment by schildbach** [comment]
Thanks a lot, Peter!
